app/lib/client.server.ts

```diff
@@ -0,0 +1,179 @@
+import { createRecord, upsertRecord, deleteRecord, getChangeStatus } from './dns.server';
+import type { RecordType } from './dns.server';
+import {
+  createRecordInDB,
+  updateRecordInDB,
+  deleteRecordInDB,
+  findRecordById,
+  doesRecordExistInDB,
+} from '../models/record.server';
+import { RecordStatus } from '@prisma/client';
+
+export const createUserDomain = async (
+  hostedZoneId: string,
```

Comment:
Throughout this PR, you're using multiple args per function.  As this number increases, positional args become more difficult to use.  Consider a function like this:

```ts
doSomething(true, true, 1, 5, false, true);
```

A better approach is to pass a single `Object` and name all the params.

Another suggestion I have: provide a default value for `hostedZoneId` using whatever is in the environment.  In testing, you can override it, but in normal code, we can omit passing it on the Object of args, and it will do the right thing.  Do this throughout this PR.

Reply:
I wanted to ask if `hostedZone` could be multiple. I guess it won't be our concern. Will be updated accordingly.